src/compiler/rust.rs

```diff
@@ -317,61 +317,193 @@ impl<T> Compiler<T> for Rust
     }
 }
 
+macro_rules! make_os_string {
+    ($( $v:expr ),*) => {{
+        let mut s = OsString::new();
+        $(
+            s.push($v);
+        )*
+        s
+    }};
+}
+
+#[derive(Clone, Debug, PartialEq)]
+struct ArgLinkLibrary {
```

Comment:
All of these could stand to have a one-line descriptive comment, like "Represents -l [KIND=]PATH link arguments to rustc".